lib/nvector_eigen.h

```diff
@@ -0,0 +1,528 @@
+#ifndef MEPBM_NVECTOR_EIGEN_H
+#define MEPBM_NVECTOR_EIGEN_H
+
+#include "sundials/sundials_nvector.h"
+#include "sundials/sundials_types.h"
+#include "eigen3/Eigen/Sparse"
+#include "eigen3/Eigen/Dense"
+#include <memory>
+#include <cmath>
+
+
+
+/**************************************************************************
+ * Functions whose memory locations are given to the N_Vector Ops structure
+ *************************************************************************/
+
+namespace NVectorOperations
+{
+  /// Returns the length of the vector
+  template <typename VectorType>
+  sunindextype
+  N_VGetLength( N_Vector v)
+  {
+    return (static_cast<VectorType*>(v->content))->size();
+  }
+
+
+
+  /// Creates a new N_Vector with the same ops field as an existing N_Vector. Allocates storage for the vector.
+  template <typename VectorType>
+  N_Vector
+  N_VClone( N_Vector w)
+  {
+    N_Vector v = N_VCloneEmpty(w);
+
+    // create memory for the vector in the new N_Vector
+    // the corresponding delete is called in N_VDestroy()
+    auto cloned = new VectorType(w->ops->nvgetlength(w));
+    v->content = cloned;
+    return v;
+  }
+
+
+
+  /// Creates a new N_Vector with the same ops field as an existing N_Vector. Does not allocate storage for the vector.
+  N_Vector
+  N_VCloneEmpty( N_Vector w)
+  {
+    N_Vector v = N_VNewEmpty();
+
+    // copy pointers to functions from existing vector
+    v->ops = w->ops;
```

Comment:
This only copies the pointer. You're almost certainly going to get into trouble with this because it's not clear to me who owns the object being pointed to.